cache/internal/profile/cache.go

```diff
@@ -0,0 +1,110 @@
+package profile
```

Comment:
I'd call it `cache` but okay :)

Reply:
It's cache with purpose only for profiles, so I decided to use the profile package.
But maybe need to consider general cache, that accept generic and works for any type and extract to it's own cache package